imsim/skycat.py

```diff
@@ -0,0 +1,374 @@
+        gamma2 = skycat_obj.get_native_attribute('shear_2')
+        if self.flip_g2:
+            gamma2 *= -1
+        kappa =  skycat_obj.get_native_attribute('convergence')
+        # Return reduced shears and magnification.
+        g1 = gamma1/(1. - kappa)    # real part of reduced shear
+        g2 = gamma2/(1. - kappa)    # imaginary part of reduced shear
+        mu = 1./((1. - kappa)**2 - (gamma1**2 + gamma2**2)) # magnification
+        return g1, g2, mu
+
+    def getDust(self, index, band='i'):
+        """
+        Return the extinction parameters for the skyCatalog object
+        corresponding to the specified index.
+
+        Parameters
+        ----------
+        index : int
+            Index of the (object_index, subcomponent) combination.
+
+        Returns
+        -------
+        (internal_av, internal_rv, galactic_av, galactic_rv)
+        """
+        skycat_obj = self.get_skycat_obj(index)
+        # For all objects, internal extinction is already part of the SED,
+        # so Milky Way dust is the only source of reddening.
+        internal_av = 0
+        internal_rv = 1.
+        MW_av_colname = f'MW_av_lsst_{band}'
+        galactic_av = skycat_obj.get_native_attribute(MW_av_colname)
+        galactic_rv = skycat_obj.get_native_attribute('MW_rv')
+        return internal_av, internal_rv, galactic_av, galactic_rv
+
+    def getObj(self, index, gsparams=None, rng=None, bandpass=None,
+               chromatic=False, exp_time=30):
+        """
+        Return the galsim object for the skyCatalog object
+        corresponding to the specified index.
+
+        Parameters
+        ----------
+        index : int
+            Index of the (object_index, subcomponent) combination.
+
+        Returns
+        -------
+        galsim.GSObject
+        """
+        obj_index, component = self._index[index]
+        skycat_obj = self.objects[obj_index]
+        sed, magnorm = self.getSED_info(index)
+        if sed is None or magnorm >= 50:
+            return None
+
+        if gsparams is not None:
+            gsparams = galsim.GSParams(**gsparams)
+
+        if skycat_obj.object_type == 'star':
+            obj = galsim.DeltaFunction(gsparams=gsparams)
+        elif (skycat_obj.object_type == 'galaxy' and
+              component in ('bulge', 'disk', 'knots')):
+            my_component = component
+            if my_component == 'knots':
+                my_component = 'disk'
+            a = skycat_obj.get_native_attribute(f'size_{my_component}_true')
+            b = skycat_obj.get_native_attribute(f'size_minor_{my_component}_true')
+            assert a >= b
+            pa = skycat_obj.get_native_attribute('position_angle_unlensed')
+            if self.flip_g2:
+                beta = float(90 - pa)*galsim.degrees
+            else:
+                beta = float(90 + pa)*galsim.degrees
+            # TODO: check if should be hlr = a. See similar note in instcat.py.
```

Comment:
I think for the skycat stuff, we should be able to figure this out, since Joanne's numbers are closer to the original simulation code.  We just need to figure out what definition that catalog uses for these values.